services/order.py

```diff
@@ -0,0 +1,47 @@
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Ticket, Order, MovieSession
+
+
+@transaction.atomic
+def create_order(tickets: list[dict],
+                 username: str, date: str | None = None) -> None:
+    user = get_user_model().objects.get(username=username)
+
+    order = Order.objects.create(user=user)
+
+    if date:
+        order.created_at = date
+        order.save()
+
+    movie_session_ids = set(ticket_data["movie_session"]
+                            for ticket_data in tickets)
+
+    movie_sessions = {
+        movie_session.id: movie_session
+        for movie_session
+        in MovieSession.objects.filter(id__in=movie_session_ids)
+    }
```

Comment:
I've written the new code but commented it because when I use only id the test fails with an error:
django.db.utils.IntegrityError: The row in table 'db_ticket' with primary key '1' has an invalid foreign key: db_ticket.movie_session_id contains a value '1' that does not have a corresponding value in db_moviesession.id.